src/node_api.cc

```diff
@@ -658,9 +658,16 @@ void napi_clear_last_error(napi_env env) {
 napi_status napi_set_last_error(napi_env env, napi_status error_code,
                                 uint32_t engine_error_code,
                                 void* engine_reserved) {
-  env->last_error.error_code = error_code;
-  env->last_error.engine_error_code = engine_error_code;
-  env->last_error.engine_reserved = engine_reserved;
+
+  if (env == nullptr) {
+    if (error_code == napi_ok) {
```

Comment:
Its better to leave in so that we don't have the case were code can get away without passing it in today, but then later fails if something changes and we do need to use it internally.